pkg/server/server.go

```diff
@@ -1004,6 +982,15 @@ func (s *Server) startPersistingHLCUpperBound(
 	return nil
 }
 
+// collectCallCounts is a middleware to collect telemetry on the call counts
+// of paths served by `inner`.
+func collectCallCounts(inner http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		telemetry.Count(fmt.Sprintf("HTTP: %s -> %s", req.Method, req.URL.EscapedPath()))
```

Comment:
I'm a bit nervous about logging URLs directly in case they contain PII or other customer data we don't want to be capturing. From a quick look at `status.proto` and `admin.proto` the URLs seem limited to containing node and range IDs but would be nice to capture the "pattern" of the URL and not the literal one being requested if we could. Not sure if that's possible.

In addition, logging URL literals could really explode the number of counters. Any concerns there? I don't know much about the storage of those and whether we need to worry about having too many keys.

Reply:
The `EscapedPath` of the `URL` should only include the path itself and not the query params. Here is an example of what it looks like on `./cockroach demo` after loading up the admin UI and clicking around on the metrics page a bit. 
<img width="479" alt="Screen Shot 2020-07-24 at 5 11 09 PM" src="https://user-images.githubusercontent.com/10788754/88435703-bf06b180-cdd0-11ea-912b-fdf0433e22fa.png">

Is there any way I could confirm that the URL paths themselves won't contain any PII? A cursory glance seems tells me that the list of endpoints here will be limited to the ones Alex listed in his API guidelines doc.

Reply:
I think it should be fine. I realized that since we do client-side telemetry logging as well which often exposes URLs we likely already make sure that those are "safe" to share.

@piyush-singh any thoughts on the discussion here? Also does this telemetry look like what you're expecting to capture?